File: c#/src/Docmost.Sdk/Models/Share.cs
namespace Docmost.Sdk.Models;

public class Share
{
    [System.Text.Json.Serialization.JsonPropertyName("id")]
    public string? Id { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("key")]
    public string? Key { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("pageId")]
    public string? PageId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("includeSubPages")]
    public string? IncludeSubPages { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("searchIndexing")]
    public string? SearchIndexing { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("creatorId")]
    public string? CreatorId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("spaceId")]
    public string? SpaceId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("workspaceId")]
    public string? WorkspaceId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("createdAt")]
    public string? CreatedAt { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("updatedAt")]
    public string? UpdatedAt { get; set; }

}

File: go/docmost/models/share.go
package models

// Share from OpenAPI.
type Share struct {
	Id *string `json:"id,omitempty"`
	Key *string `json:"key,omitempty"`
	PageId *string `json:"pageId,omitempty"`
	IncludeSubPages *string `json:"includeSubPages,omitempty"`
	SearchIndexing *string `json:"searchIndexing,omitempty"`
	CreatorId *string `json:"creatorId,omitempty"`
	SpaceId *string `json:"spaceId,omitempty"`
	WorkspaceId *string `json:"workspaceId,omitempty"`
	CreatedAt *string `json:"createdAt,omitempty"`
	UpdatedAt *string `json:"updatedAt,omitempty"`
}
